dss/operations/checkout.py

```diff
@@ -0,0 +1,188 @@
+                bundle_internal_status = list()
+
+                for _file in bundle_manifest['files']:
+                    temp = collections.defaultdict(blob_checkout=False, bundle_checkout=False, should_be_cached=False)
+                    bundle_key = f'bundles/{uuid}.{version}/{_file["name"]}'
+                    blob_key = compose_blob_key(_file)
+
+                    blob_status = self._verify_blob_existance(self.handle, self.bucket, blob_key)
+                    if blob_status:
+                        temp['blob_checkout'] = True
+                    if bundle_key in checkout_bundle_contents:
+                        temp['bundle_checkout'] = True
+                    if cache_flow.should_cache_file(_file['content-type'], _file['size']):
+                        temp['should_be_cached'] = True
+
+                    for x in ['name', 'uuid', 'version']:
+                        temp.update({x: _file[x]})
+                    bundle_internal_status.append(temp)
+                checkout_status[_key] = bundle_internal_status
+            elif FILE_PREFIX in _key:
+                temp = collections.defaultdict(blob_checkout=False, should_be_cached=False)
+                file_metadata = self._verify_get(self.handle, self.replica.bucket, _key)
+                blob_key = compose_blob_key(file_metadata)
+                blob_status = self._verify_blob_existance(self.handle, self.bucket, blob_key)
+                if blob_status:
+                    temp['blob_checkout'] = True
+                if cache_flow.should_cache_file(_file['content-type'], _file['size']):
+                    temp['should_be_cached'] = True
+
+                for x in ['name', 'uuid', 'version']:
+                    temp.update({x: _file[x]})
+                checkout_status[_key] = collections.defaultdict(uuid=temp)
+        print(json.dumps(checkout_status, sort_keys=True))
+        return checkout_status  # action_handler does not really use this, its just testing
+
+
+@checkout.action("start",
+                 arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
+                            "--keys": dict(nargs="+", help="keys to check the status of", required=True)})
+class start(CheckoutHandler):
+    def process_keys(self):
+        for _key in self.keys:
+            if DSS_BUNDLE_KEY_REGEX.match(_key):
+                uuid, version = self._parse_key(_key)
+                bundle_manifest = get_bundle_manifest(uuid=uuid, replica=self.replica, version=version)
+                self._sleepy_checkout(bundle_checkout, replica=self.replica, bundle_uuid=uuid, bundle_version=version)
+                for _files in bundle_manifest['files']:
+                    blob_path = compose_blob_key(_files)
+                    self._sleepy_checkout(file_checkout, replica=self.replica,
+                                          file_metadata=_files, blob_path=blob_path)
+            elif FILE_PREFIX in _key:
```

Comment:
Should be `elif _key.startswith(FILE_PREFIX)`.  Odds are low that `"files"` might be a part of the `uuid`, but it's not impossible.

Reply:
done